homeassistant/components/luftdaten/__init__.py

```diff
@@ -102,6 +102,14 @@
     from luftdaten import Luftdaten
     from luftdaten.exceptions import LuftdatenError
 
+    if ('{0}'.format(config_entry.data[CONF_SENSOR_ID]) in
+            duplicate_stations(hass) and config_entry.source == SOURCE_IMPORT):
+        _LOGGER.warning("Removing duplicate sensors for station %s",
+                        config_entry.data[CONF_SENSOR_ID])
+        hass.async_create_task(hass.config_entries.async_remove(
```

Comment:
This is not a good solution, because now on each restart it will be imported and then removed right away? that's wrong.

Reply:
Yes. Instead we should probably check for duplicate ids before adding an entry so that we never have duplicate ids in added entries.

Reply:
We already check for that though. I guess this is only for cleaning up the bad history.

Reply:
The (fixed) code in `async_setup` will prevent future duplicates. This is for cleaning up the mess. Without the cleanup we will continue to get blocked from the luftdaten servers, effectively breaking this component.